woocommerce/woocommerce-ios
# My Store Analytics: Custom range redacted view information

Closes: #12162


## Description


This PR adds an informational alert to explain why visitors and conversion information are redacted from custom range tab.

## Testing instructions


1. Start with a WordPress.com or Jetpack-connected site,
2. On the analytics area on My Store, go to the Custom Range tab. If you don't have a Custom Range tab, make one first using the calendar-plus icon,
3. When the Custom Range tab, confirm that the Visitors and Conversion area at the top (below revenue) is redacted,
4. Tap the redacted view for both Visitors and Conversion, and ensure the alert like on the screenshot below is shown.
5. Select a value in the graph, so that now Visitors and Conversion is showing some number. Tap them, ensure the alert is not shown.
6. Switch to a Jetpack CP site,
7. Tap the redacted view for both Visitors and Conversion, and ensure the alert is not shown.
8. Switch to a non-Jetpack site,
9. Tap the view for both Visitors and Conversion, and ensure the alert is not shown.

## screenshot & video


<img src="https://github.com/woocommerce/woocommerce-ios/assets/266376/c714a314-5b8b-4f95-94e9-169aad88abe5" width="42%" /> 

https://github.com/woocommerce/woocommerce-ios/assets/266376/950ab306-bdbf-4d81-986a-cdb60946b297  


---

### Additional case: handling for less than 2 days custom range.

When custom date range is in less than 2 days, we want to add this behavior:
1. When custom range is first created (in deselected state), show visits data.
2. When a value is selected, then hide visits data.
3. When a value is deselected, again show the visits data.

To test this:
1. Create a custom date range that is only 1 day apart,
2. When created, ensure that Visitors and Conversion are showing value.
3. Select a value in the graph, ensure that Visitors and Conversion are now redacted (simple redaction state).
4. Deselect the value, ensure that Visitors and Conversion are now shown again.
5. Update custom range to be 2 days apart,
6. When created, ensure that Visitors and Conversion are redacted (redacted due to custom range state),
7. Select a value in the graph, ensure that Visitors and Conversion are now shown.
8. Deselect the value, ensure that Visitors and Conversion are now redacted again.

Check video below for example:


https://github.com/woocommerce/woocommerce-ios/assets/266376/f23ff7ba-d019-49f7-8263-f1cddbd1bdd8
